src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -147,29 +147,46 @@ public FileUploadSubmissionResource(SubmissionRepository submissionRepository, R
      *
      * @param submissionId the id of the fileUploadSubmission to retrieve
      * @param correctionRound the correctionRound of the result we want to receive
+     * @param resultId the resultId for which we want do get the submission
      * @return the ResponseEntity with status 200 (OK) and with body the fileUploadSubmission, or with status 404 (Not Found)
      */
     @GetMapping("/file-upload-submissions/{submissionId}")
     @PreAuthorize("hasAnyRole('TA','INSTRUCTOR','ADMIN')")
     public ResponseEntity<FileUploadSubmission> getFileUploadSubmission(@PathVariable Long submissionId,
-            @RequestParam(value = "correction-round", defaultValue = "0") int correctionRound) {
+            @RequestParam(value = "correction-round", defaultValue = "0") int correctionRound, @RequestParam(value = "resultId", defaultValue = "0") long resultId) {
         log.debug("REST request to get FileUploadSubmission with id: {}", submissionId);
         var fileUploadSubmission = fileUploadSubmissionService.findOne(submissionId);
         var studentParticipation = (StudentParticipation) fileUploadSubmission.getParticipation();
         var fileUploadExercise = (FileUploadExercise) studentParticipation.getExercise();
         var gradingCriteria = gradingCriterionService.findByExerciseIdWithEagerGradingCriteria(fileUploadExercise.getId());
-        fileUploadExercise.setGradingCriteria(gradingCriteria);
+
         User user = userRepository.getUserWithGroupsAndAuthorities();
-        if (!authCheckService.isAtLeastTeachingAssistantForExercise(fileUploadExercise, user)) {
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(fileUploadExercise, user)
+                || (resultId > 0 && !authCheckService.isAtLeastInstructorForExercise(fileUploadExercise, user))) {
             return forbidden();
         }
-        fileUploadSubmission = fileUploadSubmissionService.lockAndGetFileUploadSubmission(submissionId, fileUploadExercise, correctionRound);
+
+        fileUploadExercise.setGradingCriteria(gradingCriteria);
+
+        if (resultId > 0) {
```

Comment:
Please add some discussion about the logic of this REST call to the JavaDoc above. In particular that the resultId is used first. In case it is not specified, the correction round is used. In case the correction round is not specified, the first correction round is assumed

Reply:
```suggestion
        if (resultId != null) {
```

Reply:
Done